### PR TITLE
E2E test for preemption and fungibility

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

E2E version of #885

This tests that a job can run in a different flavor after being preempted.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
I would like to run the E2E tests a few times

/assign @trasc

File: test/util/util.go [hunk 1]
```diff
@@ -80,17 +80,14 @@ func DeleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace)
 	if ns == nil {
 		return nil
 	}
-	err := c.DeleteAllOf(ctx, &batchv1.Job{}, client.InNamespace(ns.Name), client.PropagationPolicy(metav1.DeletePropagationBackground))
-	if err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
+	DeleteJobsInNamespace(ctx, c, ns)
```
Comment (trasc) [inline test/util/util.go:83]:
```suggestion
	err := DeleteJobsInNamespace(ctx, c, ns)
	if err != nil && !apierrors.IsNotFound(err) {
		return err
	}
```

Comment (alculquicondor) [reply test/util/util.go]: no need to check for IsNotFound anymore, but yes, I forgot to check if there is an error :)

File: test/util/util.go [hunk 2]
```diff
@@ -100,6 +97,14 @@ func DeleteNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace)
 	return nil
 }
 
+func DeleteJobsInNamespace(ctx context.Context, c client.Client, ns *corev1.Namespace) error {
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply test/util/util.go]: Done

**Comment by alculquicondor** [comment]
E2E tests passed 1/5

**Comment by alculquicondor** [comment]
E2E tests passed 2/5

**Comment by alculquicondor** [comment]
E2E tests passed 4/5

/test pull-kueue-test-e2e-main-1-24
/test pull-kueue-test-e2e-main-1-25
/test pull-kueue-test-e2e-main-1-26
/test pull-kueue-test-e2e-main-1-27

**Comment by tenzen-y** [comment]
/lgtm
feel free `/hold cancel`.

**Pull request merged**